java/ql/lib/semmle/code/java/frameworks/apache/Codec.qll

```diff
@@ -0,0 +1,135 @@
+        "org.apache.commons.codec.language.bm;Rule$PhonemeExpr;true;getPhonemes;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule$PhonemeList;false;PhonemeList;(List);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;Rule;(String,String,String,PhonemeExpr);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;Rule;(String,String,String,PhonemeExpr);;Argument[3];Argument[-1];taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;getLContext;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;getPattern;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;getPhoneme;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language.bm;Rule;true;getRContext;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language;Caverphone;true;caverphone;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language;DoubleMetaphone$DoubleMetaphoneResult;true;getAlternate;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language;DoubleMetaphone$DoubleMetaphoneResult;true;getPrimary;();;Argument[-1];ReturnValue;taint;",
+        "org.apache.commons.codec.language;Metaphone;true;metaphone;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language;Nysiis;true;nysiis;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language;RefinedSoundex;true;RefinedSoundex;(String);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language;RefinedSoundex;true;RefinedSoundex;(char[]);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language;RefinedSoundex;true;soundex;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.language;Soundex;true;Soundex;(String);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language;Soundex;true;Soundex;(String,boolean);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language;Soundex;true;Soundex;(char[]);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.language;Soundex;true;soundex;(String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;BCodec;true;BCodec;(Charset);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.net;BCodec;true;BCodec;(Charset,CodecPolicy);;Argument[0];Argument[-1];taint;",
+        "org.apache.commons.codec.net;BCodec;true;encode;(String,Charset);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;BCodec;true;encode;(String,Charset);;Argument[1];ReturnValue;taint;",
+        "org.apache.commons.codec.net;BCodec;true;encode;(String,String);;Argument[0];ReturnValue;taint;",
+        "org.apache.commons.codec.net;BCodec;true;getCharset;();;Argument[-1];ReturnValue;taint;",
```

Comment:
Missing [a bunch](https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/net/BCodec.html) of `encode` and `decode` methods.